lib/mirage.ml

```diff
@@ -1399,393 +1395,381 @@ let configure_main_xl ?substitutions ext i =
+        append fmt "ETH0=$(xe network-list bridge=xenbr0 params=uuid --minimal)";
+        append fmt "VIF=$(xe vif-create vm-uuid=$VM network-uuid=$ETH0 device=0)";
+        append fmt "echo Atting block device and making it bootable";
+        append fmt "VBD=$(xe vbd-create vm-uuid=$VM vdi-uuid=$VDI device=0)";
+        append fmt "xe vbd-param-set uuid=$VBD bootable=true";
+        append fmt "xe vbd-param-set uuid=$VBD other-config:owner=true";
+        List.iter (fun b ->
+            append fmt "echo Uploading data VDI %s" b.filename;
+            append fmt "echo VDI=$VDI";
+            append fmt "SIZE=$(stat --format '%%s' %s/%s)" root b.filename;
+            append fmt "POOL=$(xe pool-list params=uuid --minimal)";
+            append fmt "SR=$(xe pool-list uuid=$POOL params=default-SR --minimal)";
+            append fmt "VDI=$(xe vdi-create type=user name-label='%s' \
+                        virtual-size=$SIZE sr-uuid=$SR)" b.filename;
+            append fmt "xe vdi-import uuid=$VDI filename=%s/%s" root b.filename;
+            append fmt "VBD=$(xe vbd-create vm-uuid=$VM vdi-uuid=$VDI device=%d)"
+              b.number;
+            append fmt "xe vbd-param-set uuid=$VBD other-config:owner=true";
+          ) (Hashtbl.fold (fun _ v acc -> v :: acc) all_blocks []);
+        append fmt "echo Starting VM";
+        append fmt "xe vm-start uuid=$VM";
+        R.ok ()) ()
+  with
+  | Ok b -> b
+  | Error _ -> R.error_msg "couldn't open output channel for xe file"
+
+
+let clean_main_xe ~name = Bos.OS.File.delete Fpath.(v name + "xe")
+
+let configure_makefile ~target ~opam_name =
+  let open Codegen in
+  let file = Fpath.(v "Makefile") in
+  match
+    Bos.OS.File.with_oc file (fun oc () ->
+        let fmt = Format.formatter_of_out_channel oc in
+        append fmt "# %s" (generated_header ());
+        newline fmt;
+        let starget = Fmt.to_to_string Mirage_key.pp_target target in
+        append fmt "OPAM  = opam\n\
+                    \n\
+                    .PHONY: all depend clean build\n\
+                    all:: build\n\
+                    \n\
+                    depend::\n\
+                    \t$(OPAM) pin add --no-action --yes %s .\n\
+                    \t$(OPAM) install --yes --deps-only %s\n\
+                    \t$(OPAM) pin remove --no-action %s\n\
+                    \n\
+                    build::\n\
+                    \tmirage build -t %s\n\
+                    \n\
+                    clean::\n\
+                    \tmirage clean -t %s\n"
+          opam_name opam_name opam_name starget starget;
+        newline fmt;
+        append fmt "-include Makefile.user";
+        R.ok ()) ()
+  with
+  | Ok b -> b
+  | Error _ -> R.error_msg "couldn't open output channel for Makefile"
+
+let clean_makefile () = Bos.OS.File.delete Fpath.(v "Makefile")
+
+let configure_myocamlbuild () = Bos.OS.File.write Fpath.(v "myocamlbuild.ml") ""
```

Comment:
to avoid getting `ocamlbuild` error that it might be not an OCaml project and called with `-r` (see https://github.com/ocaml/ocamlbuild/blob/756fe48e76a874e5d7e20de637024c0379355d1b/src/options.ml#L373), discussion at https://github.com/mirage/mirage/pull/537 and https://github.com/mirage/mirage/pull/691#issuecomment-262559815.  But sure, that should be guarded.